standalone-examples/src/main/java/io/pravega/example/consolerw/ConsoleReader.java

```diff
@@ -30,73 +49,237 @@
+     * all the readers belonging to this group will consume the events only within the defined boundaries.
+     *
+     * @param config Configuration for a {@link ReaderGroup} that will contain read boundaries in that {@link Stream}.
+     */
+    private void readBasedOnStreamCuts(ReaderGroupConfig config) {
+        final String readerGroup = UUID.randomUUID().toString().replace("-", "");
         try (ReaderGroupManager readerGroupManager = ReaderGroupManager.withScope(scope, controllerURI)) {
-            readerGroupManager.createReaderGroup(readerGroup, readerGroupConfig, Collections.singleton(streamName));
+            // Create a reader group using the configuration with the defined StreamCut boundaries.
+            readerGroupManager.createReaderGroup(readerGroup, config);
         }
 
         try (ClientFactory clientFactory = ClientFactory.withScope(scope, controllerURI);
-             EventStreamReader<String> reader = clientFactory.createReader("reader",
-                                                                           readerGroup,
-                                                                           new JavaSerializer<String>(),
-                                                                           ReaderConfig.builder().build())) {
-            System.out.format("******** Reading events from %s/%s%n", scope, streamName);
-            EventRead<String> event = null;
+             EventStreamReader<String> reader = clientFactory.createReader("streamcut-reader",
+                     readerGroup, new JavaSerializer<>(), ReaderConfig.builder().build())) {
+
+            // The reader(s) will only read and display events within the StreamCut boundaries defined.
+            output("StreamCuts: Bounded processing example in stream %s/%s%n", scope, streamName);
+            output("Starting boundary for readers: %s (UnboundedStreamCut represents the head of the stream)%n",
+                    config.getStartingStreamCuts().get(Stream.of(scope, streamName)));
+            output("Terminal boundary for readers: %s (UnboundedStreamCut represents the tail of the stream)%n",
+                    config.getEndingStreamCuts().get(Stream.of(scope, streamName)));
+
+            EventRead<String> event;
             do {
-                try {
-                    event = reader.readNextEvent(READER_TIMEOUT_MS);
-                    if(event != null) {
-                        System.out.format("'%s'%n", event.getEvent());
-                    }
-                } catch (ReinitializationRequiredException e) {
-                    //There are certain circumstances where the reader needs to be reinitialized
-                    e.printStackTrace();
+                event = reader.readNextEvent(1000);
+                if (event.getEvent() != null) {
+                    // TODO: Delete this output when issue #87 is solved to let the user see the output
```

Comment:
Please follow the Pravega guidelines for TODOs in the code.

Reply:
Done.